pkg/store/bucket.go

```diff
@@ -3201,6 +3218,18 @@ func (r *bucketChunkReader) addLoad(id chunks.ChunkRef, seriesEntry, chunk int)
 
 // load loads all added chunks and saves resulting aggrs to refs.
 func (r *bucketChunkReader) load(ctx context.Context, res []seriesEntry, aggrs []storepb.Aggr, calculateChunkChecksum bool, bytesLimiter BytesLimiter) error {
+	r.loadingChunksMtx.Lock()
+	r.loadingChunks = true
+	r.loadingChunksMtx.Unlock()
+
+	defer func() {
+		r.loadingChunksMtx.Lock()
+		r.loadingChunks = false
+		r.loadingChunksMtx.Unlock()
+
+		close(r.finishLoadingChks)
+	}()
+
```

Comment:
```suggestion
    // when done loading, signal to anyone waiting on chunks to be loaded.
	defer r.loadingChunks.Signal()
```

Could use `r.loadingCunks.Broadcast()` here if there could be multiple Go routines waiting for chunks to be loaded.